src/index.js

```diff
@@ -1,40 +1,40 @@
 'use strict'
 
-const contentType = require('content-type')
-const getRawBody = require('raw-body')
+const rawBody = (req, { limit = 1048576 } = {}) =>
+  new Promise((resolve, reject) => {
+    const chunks = []
+    let bytes = 0
+    req
+      .on('error', reject)
+      .on('end', () => resolve(Buffer.concat(chunks, bytes)))
+      .on('data', chunk => {
+        chunks.push(chunk)
+        bytes += chunk.length
+        if (bytes > limit) {
+          reject(
+            new TypeError(`body size (${bytes}) is over the limit (${limit})`)
+          )
+        }
+      })
+  })
 
 const rawBodyMap = new WeakMap()
 
-const LIMIT = '1mb'
-
-const text = async (req, { limit = LIMIT, encoding } = {}) => {
-  const type = req.headers['content-type'] || 'text/plain'
-  const length = req.headers['content-length']
-
-  const body = rawBodyMap.get(req)
-  if (body) return body
-
-  if (encoding === undefined) {
-    encoding = contentType.parse(type).parameters.charset
-  }
-
-  const buffer = await getRawBody(req, { limit, length, encoding })
-  rawBodyMap.set(req, buffer)
-  return buffer.toString()
+const buffer = async (req, opts) => {
+  let body = rawBodyMap.get(req)
+  if (body !== undefined) return body
+  body = await rawBody(req, opts)
+  rawBodyMap.set(req, body)
+  return body
 }
 
-const buffer = async (req, { limit = LIMIT } = {}) => {
-  const length = req.headers['content-length']
-
-  const body = rawBodyMap.get(req)
-  if (body) return body
-
-  const buffer = await getRawBody(req, { limit, length })
-  rawBodyMap.set(req, buffer)
-
-  return buffer
-}
+const text = (req, opts) => buffer(req, opts).then(buffer => buffer.toString())
 
 const json = (req, opts) => text(req, opts).then(JSON.parse)
 
-module.exports = { text, json, buffer }
+const urlencoded = (req, opts) =>
+  text(req, opts).then(text =>
+    Object.fromEntries(new URLSearchParams(text).entries())
+  )
```

Comment:
can this be `new URLSearchParams`? why converting into an object? The object would not support multiple values for a given key